homeassistant/components/synology_dsm/__init__.py

```diff
@@ -562,6 +594,78 @@ async def async_added_to_hass(self):
         self.async_on_remove(self._api.subscribe(self._api_key, self.unique_id))
 
 
+class SynologyDSMCoordinatorEntity(CoordinatorEntity):
+    """Representation of a Synology NAS entry."""
+
+    def __init__(
+        self,
+        api: SynoApi,
+        entity_type: str,
+        entity_info: Dict[str, str],
+        coordinator: DataUpdateCoordinator,
+    ):
+        """Initialize the Synology DSM entity."""
+        super().__init__(coordinator)
+
+        self._api = api
+        self._api_key = entity_type.split(":")[0]
+        self.entity_type = entity_type.split(":")[-1]
+        self._name = f"{api.network.hostname} {entity_info[ENTITY_NAME]}"
+        self._class = entity_info[ENTITY_CLASS]
+        self._enable_default = entity_info[ENTITY_ENABLE]
+        self._icon = entity_info[ENTITY_ICON]
+        self._unit = entity_info[ENTITY_UNIT]
+        self._unique_id = f"{self._api.information.serial}_{entity_type}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit the value is expressed in."""
+        if self.entity_type in TEMP_SENSORS_KEYS:
+            return self.hass.config.units.temperature_unit
+        return self._unit
+
```

Comment:
Is this used?

Reply:
`SynologyDSMCoordinatorEntity` should later be used as base-class for all entities in `synology_dsm` integration and therefore is a copy of `SynologyDSMEntity`

Reply:
Since it is not used now, please remove and add it back in the future when additional changes are made.  We don't want to introduce code that isn't being used into the codebase.